### PR TITLE
add apt-file

### PR DESCRIPTION
**Pull request by sunflsks**

### CODE REVIEW

**Comment by asdfugil** [comment]
iirc this requires the repository to do something

File: apt-file.mk [hunk 1]
```diff
@@ -0,0 +1,44 @@
+ifneq ($(PROCURSUS),1)
+$(error Use the main Makefile)
+endif
+
+SUBPROJECTS       += apt-file
+APT-FILE_VERSION  := 3.2.2
+DEB_APT-FILE_V    ?= $(APT-FILE_VERSION)
+
+apt-file-setup: setup
+	wget -q -nc -P $(BUILD_SOURCE) https://salsa.debian.org/apt-team/apt-file/-/archive/debian/3.2.2/apt-file-debian-$(APT-FILE_VERSION).tar.gz
```
Comment (CRKatri) [inline apt-file.mk:10]: Replace the 3.2.2 in the url with the version variable.

File: apt-file.mk [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+	$(call EXTRACT_TAR,apt-file-debian-$(APT-FILE_VERSION).tar.gz,apt-file-debian-$(APT-FILE_VERSION),apt-file)
+	mkdir -p $(BUILD_STAGE)/apt-file/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/bin
+
+ifneq ($(wildcard $(BUILD_WORK)/apt-file/.build_complete),)
+apt-file:
+	@echo "Using previously built apt-file."
+else
+apt-file: apt-file-setup
+	$(SED) -i '1s|.*|#!$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/bin/perl|g' $(BUILD_WORK)/apt-file/apt-file
+
```
Comment (CRKatri) [inline apt-file.mk:20]: Please don't have blank lines in rules, either delete them or add a tab. They break coloring in vim.

**Comment by CRKatri** [comment]
> iirc this requires the repository to do something

Which Procursus and my repo are already doing. If you'd like to add it to your repo, use [this update script](https://cameronkatri.com/Update.sh) as an example.

**Review by CRKatri — APPROVED** [review]

**Pull request merged**